Unused private member warning/error

**Describe the bug**

```
vt/src/vt/utils/memory/memory_usage.h:105:8: error: private field 'failed_' is not used [-Werror,-Wunused-private-field]
01:17:34    bool failed_ = false;
01:17:34         ^
01:17:34  1 error generated.
```

**To Reproduce**
Steps to reproduce the behavior:
1. build vt without `vt_has_sysconf`

**Expected behavior**
A clear and concise description of what you expected to happen.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Platform (please complete the following information):**
 - Supercomputer or desktop [e.g., Trinity, Sierra)
 - Compiler environment [e.g., clang-3.9, gcc8]
 - Libraries [e.g., mpich-2, openmpi)
 - Commit ID or release of VT [e.g., vt-1.0.0]

**Additional context**
Add any other context about the problem here.

--- a/src/vt/utils/memory/memory_usage.cc
+++ b/src/vt/utils/memory/memory_usage.cc
@@ -276,6 +276,7 @@ std::size_t StatM::getUsage() {
     return 0;
   }
 # else
+  (void)failed_; // Suppress unused private variable warning
   return 0;
 # endif
 }

